docs/docs/building-apps/02-app-mempool.md

```diff
@@ -35,29 +37,75 @@ Process proposal handles the validation of what is in a block, meaning that afte
 
-There are countless designs that an application developer can write for a mempool, the core team opted to provide a simple implementation of a nonce mempool. The nonce mempool is a mempool that keeps transactions from an sorted by nonce in order to avoid the issues with nonces. 
+There are countless designs that an application developer can write for a mempool, the SDK opted to provide only simple mempool implementations.
+Namely, the SDK provides the following mempools:
+
+* [Sender Nonce Mempool](#sender-nonce-mempool)
+* [Priority Nonce Mempool](#priority-nonce-mempool)
+* [No-op Mempool](#no-op-mempool)
+
+The default SDK is a [Sender Nonce Mempool](#sender-nonce-mempool), but it can be replaced by the application developer in [`app.go`](./01-app-go-v2.md):
+
+```go reference
+nonceMempool := mempool.NewSenderNonceMempool()
+mempoolOpt   := baseapp.SetMempool(nonceMempool)
+baseAppOptions = append(baseAppOptions, mempoolOpt)
+```
+
+### Sender Nonce Mempool
 
+The nonce mempool is a mempool that keeps transactions from an sorted by nonce in order to avoid the issues with nonces. 
 It works by storing the transation in a list sorted by the transaction nonce. When the proposer asks for transactions to be included in a block it randomly selects a sender and gets the first transaction in the list. It repeats this until the mempool is empty or the block is full. 
 
-### Configurations
+It can configurable by the following parameters:
 
 #### MaxTxs
 
-Its an integer value that sets the mempool in one of three modes, bounded, unbounded, or disabled.
+It is an integer value that sets the mempool in one of three modes, *bounded*, *unbounded*, or *disabled*.
+
+* **negative**: Disabled, mempool does not insert new transaction and return early.
+* **zero**: Unbounded mempool has no transaction limit and will never fail with `ErrMempoolTxMaxCapacity`.
+* **positive**: Bounded, it fails with `ErrMempoolTxMaxCapacity` when `maxTx` value is the same as CountTx()
+
+#### Seed
+
+Set the seed for the random number generator used to select transactions from the mempool.
+
+### Priority Nonce Mempool
+
+The [priority nonce mempool](https://github.com/cosmos/cosmos-sdk/blob/main/types/mempool/priority_nonce_spec.md) is a mempool implementation that stores txs in a partially ordered set by 2 dimensions:
+
+* priority
+* sender-nonce (sequence number)
+
+Internally it uses one priority ordered [skip list](https://pkg.go.dev/github.com/huandu/skiplist) and one skip list per sender ordered by sender-nonce (sequence number). When there are multiple txs from the same sender, they are not always comparable by priority to other sender txs and must be partially ordered by both sender-nonce and priority.
+
+It can configurable by the following parameters:
+
+#### Callback
+
+Allow to set a callback to be called when a transaction is read from the mempool.
+
+### No-op Mempool
```

Comment:
I believe one of you is creating this one, right? So, I think it makes sense to already add it here.

Reply:
We haven't yet, but I recall some discussion about this.  Is there an issue for it?

Reply:
I don't think there is yet actually.

Reply:
I was thinking maybe we default to noop mempool instead of sender nonce and have users have it be explicit. Cc @alexanderbez

Reply:
Updated the docs to reflect that, but the implementation must still follow.